### PR TITLE
fix: sign out when user token invalid

### PR DESCRIPTION
**Pull request by Yuan325**
when the user token is invalid, signout to clear the session instead of just clearning session['user_info'].

### CODE REVIEW

File: llm_demo/orchestrator/.orchestrator.py.swp
Comment (averikitsch) [inline llm_demo/orchestrator/.orchestrator.py.swp:1]: Is this needed?

File: llm_demo/orchestrator/orchestrator.py [hunk 1]
```diff
@@ -66,20 +66,22 @@ def set_user_session_header(self, uuid: str, user_id_token: str):
...
     def get_user_id_token(self, uuid: str) -> Optional[str]:
-        user_session = self.get_user_session(uuid)
-        if user_session.client and "User-Id-Token" in user_session.client.headers:
-            token = user_session.client.headers["User-Id-Token"]
-            parts = str(token).split(" ")
-            if len(parts) != 2 or parts[0] != "Bearer":
-                raise Exception("Invalid ID token")
-            return parts[1]
+        if self.user_session_exist(uuid):
+            user_session = self.get_user_session(uuid)
```
Comment (averikitsch) [inline llm_demo/orchestrator/orchestrator.py:70]: nit: this could be combined with the line above

File: llm_demo/orchestrator/orchestrator.py [hunk 2]
```diff
@@ -66,20 +66,22 @@ def set_user_session_header(self, uuid: str, user_id_token: str):
...
         return None
 
     async def user_session_signout(self, uuid: str):
         """Sign out from user session. Clear and restart session."""
         user_session = self.get_user_session(uuid)
-        await user_session.close()
-        del user_session
+        if user_session:
+            await user_session.close()
+            del user_session
```
Comment (averikitsch) [inline llm_demo/orchestrator/orchestrator.py:84]: Doesn't this need to be deleted from the dictionary like `del self._user_sessions[user_session]`?

**Review by averikitsch — APPROVED** [review]

File: llm_demo/orchestrator/.orchestrator.py.swp
Comment (Yuan325) [reply llm_demo/orchestrator/.orchestrator.py.swp:1]: whoops, thanks!

**Pull request merged**